analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InfiniteRecursion.CSharp11.cs

```diff
@@ -14,6 +14,44 @@ interface InfiniteRecursion
 
     public class Addition : IAdditionOperators<Addition, Addition, Addition>
     {
-        public static Addition operator +(Addition left, Addition right) => left + right; // FN
+        public static Addition operator +(Addition left, Addition right) => left + right; // Noncompliant
```

Comment:
this is the list of the overloadable operators: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/operator-overloading#overloadable-operators

I suggest to look at them in [SharpLab](https://sharplab.io/), see what  SyntaxKinds you don't cover from them and add examples 

(it's probably overkill to add them all, however as you cover Binary, Decrement, Increment, you should cover the other ones as well)

Also, the list of the IOperator... interfaces are here: https://learn.microsoft.com/en-us/dotnet/api/system.numerics.iequalityoperators-3?view=net-7.0